subsys/net/lib/tftp/Kconfig

```diff
@@ -0,0 +1,49 @@
+# Socket TFTP Library for Zephyr
+
+# Copyright (c) 2020 InnBlue
+# SPDX-License-Identifier: Apache-2.0
+
+config TFTP_LIB
+	bool "Socket TFTP Library Support [EXPERIMENTAL]"
+	select NET_SOCKETS
+	select NET_SOCKETS_POSIX_NAMES
+	help
+	  Enable the Zephyr TFTP Library
+
+if TFTP_LIB
+
+module=TFTP
+module-dep=NET_LOG
+module-str=Log level for TFTP
+module-help=Enables tftp debug messages.
+source "subsys/net/Kconfig.template.log_config.net"
+
+config TFTPC_MAX_FILENAME_SIZE
```

Comment:
I'm sorry, but this setting doesn't make sense. You do allocate a static buffer for entire TFTP packet. Until the suggestion to let dependency-inject this buffer from app side, you should have the config option which specifies the size of this static buffer. Then max filename size then just depends on this buffer size. And for current default for buffer size of 512, the max filename size is >500 bytes.